File: ios/src/ReactNative/RNIContextMenuView/RNIContextMenuViewModule.m
//
//  RNIContextMenuViewModule.m
//  react-native-ios-context-menu
//
//  Created by Dominic Go on 11/16/21.
//

#import "React/RCTBridgeModule.h"


@interface RCT_EXTERN_MODULE(RNIContextMenuViewModule, NSObject)


RCT_EXTERN_METHOD(dismissMenu: (nonnull NSNumber)node
                  // promise blocks -----------------------
                  resolve: (RCTPromiseResolveBlock *)resolve
                  reject : (RCTPromiseRejectBlock  *)reject);

RCT_EXTERN_METHOD(provideDeferredElements: (nonnull NSNumber)node
                  deferredID: NSString
                  menuItems : NSArray
                  // promise blocks -----------------------
                  resolve: (RCTPromiseResolveBlock *)resolve
                  reject : (RCTPromiseRejectBlock  *)reject);

@end

File: src/native_components/RNIContextMenuView/RNIContextMenuViewModule.ts
import { requireNativeModule } from 'expo-modules-core';
import type { NotifyOnComponentWillUnmount } from 'react-native-ios-utilities';

import type { MenuElementConfig } from '../../types/MenuConfig';


interface RNIContextMenuViewModule  {
  presentMenu(
    node: number,
  ): Promise<void>;

  dismissMenu(
    node: number,
  ): Promise<void>;

  showAuxiliaryPreviewAsPopover(
    node: number,
  ): Promise<void>;

  provideDeferredElements(
    node: number,
    args: {
      deferredID: string, 
      menuItems: MenuElementConfig[]
    }
  ): Promise<void>;
};

export const RNIContextMenuViewModule: RNIContextMenuViewModule = 
  requireNativeModule('RNIContextMenuView');


  
